### PR TITLE
Gardening: bail without erroring on forks without Slack tokens

### PR DESCRIPTION
**Pull request by jeherve**
#### Changes proposed in this Pull Request:

The Notify Design and Notify Editorial tasks run on pull_request events, and require access to secret tokens set in the settings of the repo. Those secrets will not be available when the pull requests are created from a fork.

In those scenarios, let's bail without throwing an error.
    
#### Jetpack product discussion

* Internal reference: p1620953270220300-slack-calypso

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* Not much to test on this PR, since it is not created from a fork.

### CODE REVIEW

File: projects/github-actions/repo-gardening/changelog/fix-failing-task-external-contributors [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: fixed
+
+Tasks relying on Slack running on pull_request events: bail early wwithout failing when PR was opened from a fork.
```
Comment (Aurorum) [inline projects/github-actions/repo-gardening/changelog/fix-failing-task-external-contributors:4]:
```suggestion
Tasks relying on Slack running on pull_request events: bail early without failing when PR was opened from a fork.
```

File: projects/github-actions/repo-gardening/src/tasks/notify-design/index.js [hunk 2]
```diff
@@ -67,10 +67,19 @@ async function hasDesignInputRequestedLabel( octokit, owner, repo, number ) {
...
-	const { number, repository } = payload;
+	const { number, pull_request, repository } = payload;
+	const { head, base } = pull_request;
 	const { owner, name: repo } = repository;
 	const ownerLogin = owner.login;
 
+	// Bail early for OSS contributors. They do not have access to Slack tokens.
+	if ( head.repo.full_name !== base.repo.full_name ) {
+		debug(
+			`notify-design: Design input will not be requerested for PR #${ number }, it was created from a fork. Aborting.`
```
Comment (Aurorum) [inline projects/github-actions/repo-gardening/src/tasks/notify-design/index.js:78]:
```suggestion
			`notify-design: Design input will not be requested for PR #${ number }, it was created from a fork. Aborting.`
```

File: projects/github-actions/repo-gardening/src/tasks/notify-editorial/index.js [hunk 3]
```diff
@@ -67,10 +67,19 @@ async function hasEditorialInputRequestedLabel( octokit, owner, repo, number ) {
...
-	const { number, repository } = payload;
+	const { number, pull_request, repository } = payload;
+	const { head, base } = pull_request;
 	const { owner, name: repo } = repository;
 	const ownerLogin = owner.login;
 
+	// Bail early for OSS contributors. They do not have access to Slack tokens.
+	if ( head.repo.full_name !== base.repo.full_name ) {
+		debug(
+			`notify-editorial: Design input will not be requerested for PR #${ number }, it was created from a fork. Aborting.`
```
Comment (Aurorum) [inline projects/github-actions/repo-gardening/src/tasks/notify-editorial/index.js:78]:
```suggestion
			`notify-editorial: Editorial input will not be requested for PR #${ number }, it was created from a fork. Aborting.`
```

**Review by Aurorum — COMMENTED** [review]
Couple of (I think) typos which I just spotted below!

**Comment by jeherve** [comment]
@Aurorum Thanks for the review, good catch!

**Comment by anomiex** [comment]
Could we switch these to run on `pull_request_target` instead of `pull_request` instead? That would allow them to access the token.

**Comment by jeherve** [comment]
> Could we switch these to run on `pull_request_target` instead of `pull_request` instead? That would allow them to access the token.

That's a good point. I suppose we could. :) I gave this a try: c6f1925d597ca8674c3c142b2eb6c7cbd74d3887

**Review by anomiex — APPROVED** [review]

**Pull request merged**